src/thunks/searchThunk.js

```diff
@@ -4,51 +4,49 @@ const treeherderBaseURL = 'https://treeherder.mozilla.org';
 
 export const fetchRecentRevisions = createAsyncThunk(
   'search/fetchRecentRevisions',
-  async (repository, { rejectWithValue }) => {
-    let response;
-    try {
-      response = await fetch(
-        `${treeherderBaseURL}/api/project/${repository}/push/`,
-      )
-        .then((res) => res.json())
-        .then((res) => res.results);
-    } catch (err) {
-      return rejectWithValue(err.response.data);
+  async (repository, { getState, requestId }) => {
+    const { currentRequestID, loading } = getState().search;
+    if (loading !== 'pending' || requestId !== currentRequestID) {
+      return [];
     }
+    const response = await fetch(
+      `${treeherderBaseURL}/api/project/${repository}/push/`,
+    )
+      .then((res) => res.json())
+      .then((res) => res.results);
     return response;
   },
 );
 
 export const fetchRevisionByID = createAsyncThunk(
   'search/fetchRevisionByID',
-  async ({ repository, search }, { rejectWithValue }) => {
-    let response;
-    try {
-      response = await fetch(
-        `${treeherderBaseURL}/api/project/${repository}/push/?revision=${search}`,
-      )
-        .then((res) => res.json())
-        .then((res) => res.results);
-    } catch (err) {
-      return rejectWithValue(err.response.data);
+  async ({ repository, search }, { getState, requestId }) => {
+    const { currentRequestID, loading } = getState().search;
+    if (loading !== 'pending' || requestId !== currentRequestID) {
+      return [];
     }
+    const response = await fetch(
+      `${treeherderBaseURL}/api/project/${repository}/push/?revision=${search}`,
```

Comment:
(for another patch)

I think you should either use `encodeURIComponent` or the `URLSearchParams` object to make sure this `search` value isn't malicious.

Also not sure where `treeherderBaseURL` and `repository` come from, but you might need to use `encodeURIComponent` for them too, if they come from some untrusted place.

These are probably not a very big deal in this case because this is just a `get` operation, but better do it to be safe :)

Obviously the same comment applies to the other fetch commands.

Reply:
the URL is a `const` hardcoded in this same file, the `repository` is coming from the store. it is set when a user selects a repository from the dropdown

Reply:
I think it should be done for the `search` at least. This might even bring bugs depending on the characters in the search, such as +, %, or anything that has a special meaning in a URL.

For `repository` my gut feeling is that if you don't ensure that the value is some specific hardcoded value (with an allowlist for example) it should also be escaped but it's likely less important.

Reply:
currently the `fetch` will only occur if the search value matches either an email or 40 character hash (alphanumeric basically) which I will be extending to include a short hash (12 character alphanumeric string), since I now know that the treeherder API accepts short hashes (#68)

https://github.com/mozilla/perfcompare/blob/master/src/utils/searchViewHelper.js#L14-L23